plugin/al/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,348 @@
+
+- Schema based translators:
+  - You will need to update the translators to be [`PrimUpdater` plugin](https://github.com/Autodesk/maya-usd/blob/release/v0.27.0/lib/mayaUsd/fileio/primUpdaterRegistry.h#L34-L60)
+
+- "assettype" metadata based translators (Python translators):
+  - You will need to run them manually
+
+See [translator system](#translator-plugin-system) above for more details.
+
+#### `AL_usdmaya_ProxyShapeFindLoadable`
+
+No equivalent command in ADSK_USDMaya, recommends finding layers with native USD API.
+
+#### `AL_usdmaya_ListTranslators`
+
+This command is part of the [Translator Plugin System](#translator-plugin-system), there is no equivalent command in ADSK_USDMaya.
+
+#### `AL_usdmaya_ProxyShapeImportAllTransforms`
+
+This command creates a Maya transform hierarchy next to the proxy shape node that mirrors the stage hierarchy. There is no equivalent command in ADSK_USDMaya, which recommends creating a transform chain with native Maya API.
+
+#### `AL_usdmaya_ProxyShapeRemoveAllTransforms`
+
+No equivalent command in ADSK_USDMaya, recommends removing the transform chain with native Maya API.
+
+#### `AL_usdmaya_ProxyShapeImportPrimPathAsMaya`
+
+No equivalent command in ADSK_USDMaya, recommends managing transform nodes with native Maya API.
+
+### AL Event System and Callbacks Commands
+
+AL_USDMaya implemented a custom event system (see [Core Event System](https://github.com/Autodesk/maya-usd/blob/release/v0.27.0/plugin/al/lib/AL_USDMaya/AL/docpages/docs_events.h) and [Maya Event System](https://github.com/Autodesk/maya-usd/blob/release/v0.27.0/plugin/al/lib/AL_USDMaya/AL/docpages/docs_mayaevents.h)) to group and prioritize global and per-node callback, there is no equivalent in ADSK_USDMaya, however, all of them can be easily reimplemented by native Maya [MMessage](https://help.autodesk.com/view/MAYAUL/2024/ENU/?guid=MAYA_API_REF_cpp_ref_class_m_message_html).
+
+Commands below have no equivalent in ADSK_USDMaya, we recommend to reimplement with native Maya MMessage API and manage the callbacks and events separately (in your code):
+
+- `AL_usdmaya_command_Callback`
+- `AL_usdmaya_command_ListEvent`
+- `AL_usdmaya_ListEvents`
+- `AL_usdmaya_ListCallbacks`
+- `AL_usdmaya_Callback`
+- `AL_usdmaya_DeleteCallbacks`
+- `AL_usdmaya_CallbackQuery`
+- `AL_usdmaya_Event`
+- `AL_usdmaya_EventQuery`
+- `AL_usdmaya_EventLookup`
+- `AL_usdmaya_TriggerEvent`
+
+### Prim and Layer manipulation commands
+
+#### `AL_usdmaya_LayerManager`
+
+This command allows you to query dirty layers currently managed by LayerManager in AL_USDMaya, there is no equivalent command in ADSK_USDMaya, we recommend querying dirty layers via native USD API.
+
+Following commands are wrappers around USD APIs, there is no equivalent commands in ADSK_USDMaya, recommend switching over to native USD APIs:
+
+- `AL_usdmaya_ChangeVariant`
+- `AL_usdmaya_ActivatePrim`
+- `AL_usdmaya_CreateUsdPrim`
+- `AL_usdmaya_LayerCreateLayer`
```

Comment:
Not sure if this is the same, but anonymous Layers can be created through the mayaUSDLayerEditor command:
`mayaUsdLayerEditor -edit -addAnonymous  "anonymousLayer2" "anon:000002B6CB1D8DC0:anonymousLayer1";`